convert/convert.go

```diff
@@ -62,11 +83,48 @@ type Converter interface {
 	writeFile(io.WriteSeeker, llm.KV, []llm.Tensor) error
 }
 
+func ConvertAdapter(fsys fs.FS, ws io.WriteSeeker, baseLayer *llm.GGML) error {
+	bts, err := fs.ReadFile(fsys, "adapter_config.json")
+	if err != nil {
+		return err
+	}
+
+	var p Parameters
+	if err := json.Unmarshal(bts, &p); err != nil {
+		return err
+	}
+
+	ts, err := parseTensors(fsys)
+	if err != nil {
+		return err
+	}
+
+	baseKV := baseLayer.KV()
+	arch, ok := baseKV["general.architecture"]
+	if !ok {
+		return errors.New("architecture not set for the base model")
+	}
+
+	var conv Converter
```

Comment:
this should not reuse Converter. instead create an AdapterConvert (and possibly rename Converter to ModelConverter)